devsecops/artifact-registry-vulnerability-scanning/cloud-function/index.js

```diff
@@ -0,0 +1,49 @@
+const { ContainerAnalysisClient } = require('@google-cloud/containeranalysis');
+const functions = require('@google-cloud/functions-framework');
+const { processVulnerability } = require('./src/process-vunerability.js');
+
+const containerAnalysisClient = new ContainerAnalysisClient();
+
+exports.imageVulnPubSubHandler = async (event, context) => {
+  try {
+    // Decode the base64-encoded pubsub data and parse it as JSON (this is the metadata)
+    const decodedData = Buffer.from(event.data, 'base64').toString('utf-8');
+    const data = JSON.parse(decodedData);
+
+    // Get bucketName
+    const bucketName =
+      process.env.BUCKET_NAME || 'safe-inputs-devsecops-outputs-for-dashboard';
+    if (!bucketName) {
+      console.error('Bucket name not set');
+      return;
+    }
+
+    // Get the occurrence info using the Grafeas client (from data.name)
+    const occurrenceName = data.name;
+    const grafeasClient = containerAnalysisClient.getGrafeasClient();
+    const [occurrence] = await grafeasClient.getOccurrence({
+      name: occurrenceName,
+    });
+
+    // We're only interested in vunerability types - filter for these and process (which includes saving to bucket)
+    //  Using others for logging - can remove in future.
```

Comment:
I assume you want the logs for other cases on for the initial merge, and then you'll turn them off after validating everything works? If there aren't an excessive amount of logs, I guess we could leave them on, but I think they'll be pretty noisy and low value.

Reply:
No, I agree we'll turn them off, there's quite a few of them and the files are appearing in the bucket much more quickly than before so no need to give assurance that the function processing anymore.